credentials/src/main/kotlin/web5/sdk/credentials/PresentationExchange.kt

```diff
@@ -20,40 +28,77 @@ public object PresentationExchange {
    *
-   * @param presentationDefinition The Presentation Definition to be evaluated.
-   * @return `true` if the Presentation Definition is satisfied, `false` otherwise.
-   * @throws UnsupportedOperationException if certain features like Submission Requirements or Field Filters are not implemented.
+   * @param vcJwt The VC in JWT format.
+   * @param presentationDefinition The Presentation Definition to be satisfied.
    */
   public fun satisfiesPresentationDefinition(
-    credential: VerifiableCredential,
+    vcJwt: String,
     presentationDefinition: PresentationDefinitionV2
-  ): Boolean {
+  ) {
+    val vc = JWTParser.parse(vcJwt) as SignedJWT
+
     if (!presentationDefinition.submissionRequirements.isNullOrEmpty()) {
       throw UnsupportedOperationException(
         "Presentation Definition's Submission Requirements feature is not implemented"
       )
     }
 
-    return presentationDefinition.inputDescriptors
+    presentationDefinition.inputDescriptors
       .filter { !it.constraints.fields.isNullOrEmpty() }
-      .all { inputDescriptorWithFields ->
-        val requiredFields = inputDescriptorWithFields.constraints.fields!!.filter { it.optional != true }
-
-        var satisfied = true
-        for (field in requiredFields) {
-          // we ignore field filters
-          if (field.filterSchema != null) {
-            throw UnsupportedOperationException("Field Filter is not implemented")
-          }
-
-          if (field.path.any { path -> credential.getFieldByJsonPath(path) == null }) {
-            satisfied = false
-            break
-          }
-        }
-        return satisfied
+      .forEach { inputDescriptorWithFields ->
+        validateInputDescriptorsWithFields(inputDescriptorWithFields, vc.payload)
       }
   }
+
+  /**
+   * Validates whether the input descriptors with fields in a Presentation Definition are satisfied by
+   * the payload of a verifiable credential (VC).
+   *
+   * @param inputDescriptorWithFields The input descriptor with fields to be validated.
+   * @param vcPayload The payload of the VC.
+   */
+  private fun validateInputDescriptorsWithFields(
+    inputDescriptorWithFields: InputDescriptorV2,
+    vcPayload: Payload
+  ) {
+    val requiredFields = inputDescriptorWithFields.constraints.fields!!.filter { it.optional != true }
+
+    requiredFields.forEach { field ->
+      val vcPayloadJson = JsonPath.parse(vcPayload.toString())
+        ?: throw PresentationExchangeError("Failed to parse VC $vcPayload as JsonNode")
+
+      val matchedFields = field.path.mapNotNull { path -> vcPayloadJson.read<JsonNode>(path) }
+      if (matchedFields.isEmpty()) {
+        throw PresentationExchangeError("Could not find matching field for path: ${field.path.joinToString()}")
+      }
+
+      when {
+        field.filterSchema != null -> matchedFields.any {vcSatisfiesFieldFilterSchema(it, field.filterSchema!!)}
```

Comment:
matched field could be either an array or a single value.
if array, filterSchema needs to be applied to each item in the array. If any items satisfy the schema, return true
if single value, apply schema

Reply:
matched field could also be an object